### ISSUE TITLE
Remap mouse wheel click as ERASE TILE

### ISSUE THREAD

**Issue by caiolopez** [bjorn/tiled#1068]

**Comment by bjorn** [comment]
Well, that's tricky. It would have to be added as an option, and I don't like adding such very specific options. The alternative is to allow arbitrary remapping of mouse buttons and this of course comes with its own large set of problems to solve.

I would like to make keyboard shortcuts and possibly also mouse buttons configurable at some point, but this will take a while.

**Comment by caiolopez** [comment]
Yeah, I see, a big feature with low priority for now!
Would a non-toggle erase function like "hitting del to erase the tile under the cursor" be easier/better to implement? I see that the shortcut for it isn't taken yet. That way the user can both save time by not switching back and forth from two different toggle tools every time he makes a mistake, and map it to any other input he/she sees fit using 3rd party software, if needed.
Thank you!

**Comment by bjorn** [comment]
Hmm, well the Delete shortcut is of course taken.

I wonder if there would be some way to make all tools usable temporarily. So for eraser you normally press `E`, but what if by using press-and-hold `E` it would switch back to the previous tool afterwards? I'm not currently sure what other issues that would open up, but I think it may work.

**Comment by caiolopez** [comment]
That's how the "eye dropper tool" works in photoshop. If you push and release "I", it will toggle, but if you you just hold "I" and click it will pick the clicked color and then switch back to the previous tool automatically upon key release. Works pretty well, I think.

**Comment by bjorn** [comment]
@caiolopez Ok, it's cool that Photoshop does this already since it indeed indicates that it can work and it means people could already be intuitively expecting this feature. But did they special-case this for the eye dropper or does it work more generally?

**Comment by wayfu** [comment]
It works on all tools as far as I know.

**Comment by caiolopez** [comment]
indeed, brush, eraser and everybody else work like that in Photoshop.
